src/WebJobs.Script/Description/FunctionInvokerBase.cs

```diff
@@ -84,15 +84,35 @@ public async Task<object> Invoke(object[] parameters)
 
         private static FunctionInvocationContext GetContextFromParameters(object[] parameters, FunctionMetadata metadata)
         {
+            ExecutionContext functionExecutionContext = null;
+            Binder binder = null;
+            ILogger logger = null;
+
+            for (var i = 0; i < parameters.Length; i++)
+            {
+                switch (parameters[i])
+                {
+                    case ExecutionContext fc:
+                        functionExecutionContext ??= fc;
+                        break;
+                    case Binder b:
+                        binder ??= b;
+                        break;
+                    case ILogger l:
+                        logger ??= l;
+                        break;
+                }
```

Comment:
Note that we can write this more compact, but I'm not sure everyone will find it intuitive:
```cs
for (var i = 0; i < parameters.Length; i++)
{
    _ = parameters[i] switch
    {
        ExecutionContext fc => functionExecutionContext ??= fc,
        Binder b => binder ??= b,
        ILogger l => logger ??= l,
        _ => (object)null,
    };
}
```